src/objectExplorer/objectExplorerService.ts

```diff
@@ -267,14 +267,22 @@ export class ObjectExplorerService {
                 } else {
                     // start node session
                     let promise = new Deferred<TreeNodeInfo>();
-                    await this.createSession(promise, element.connectionCredentials);
-                    let node = await promise;
-                    // If node create session failed
-                    if (!node) {
+                    try {
+                        await this.createSession(promise, element.connectionCredentials);
+                        let node = await promise;
+                        // if password failed
+                        if (!node) {
+                            const connectNode = new ConnectTreeNode(element);
+                            this._treeNodeToChildrenMap.set(element, [connectNode]);
+                            return [connectNode];
+                        }
+                    } catch (err) {
+                        // If node create session failed
                         const signInNode = new AccountSignInTreeNode(element);
```

Comment:
Do we really want to re-create the node each time? If the sign in node was already there it'd be good not to recreate it.

Reply:
This is run only once because once this node is made, the code path is different.